design/ceph/node-loss-rbd-rwo.md

```diff
@@ -0,0 +1,123 @@
+For this we'll follow the following approach which will work for all networking options(but still will not work for external cluster which nodeIps):
+
+1. Get the `volumesInUse` from the node which has the taint `out-of-service`.
+
+2. List all the pv and compare the pv `spec.volumeHandle` with the node `volumesInUse` field `volumeHandle`
+   1. Example:
+
+      Below is sample Node volumeInUse field
+
+    ```
+      volumesInUse:
+        - kubernetes.io/csi/rook-ceph.rbd.csi.ceph.com^0001-0009-rook-ceph-0000000000000002-24862838-240d-4215-9183-abfc0e9e4002
+
+      # Note: The volumeInUse naming convention are `kubernetes.io/csi/ + CSI driver name + ^ + volumeHandle`
+    ```
+
+     and the following is pv `volumeInHandle`
+
+     ```
+      volumeHandle: 0001-0009-rook-ceph-0000000000000002-24862838-240d-4215-9183-abfc0e9e4002
+    ```
+
+3. For Ceph volumes on that node:
+
+   1. If RBD PVC makes use of the rbd status API
+      1. example:
+
+    ```console
+    $ rbd status <image_name>
+    Watchers:
+    watcher=172.21.12.201:0/4225036114 client.17881 cookie=18446462598732840961
+    ```
+
+   2. If CephFS PVC uses ceph fs client ls  API
+      1.
+        ```TODO```
+
+4. Get IPs from step 3 (in above example `172.21.12.201:0/4225036114`)
+5. Blocklist the endpoint where the volumes are mounted.
+
+Example of a NetworkFence CR that the Rook operator would create when a node is lost:
+
+```yaml
+apiVersion: csiaddons.openshift.io/v1alpha1
+kind: NetworkFence
+metadata:
+  name: <name> # We will keep the name the same as the node name
+  namespace: <ceph-cluster-namespace>
+spec:
+  driver: <driver-name> # for us it will `rook-ceph.rbd.csi.ceph.com`
+  fenceState: <fence-state> # For us it will be `Fenced`
+  cidrs:
+    - <172.21.12.201:0/4225036114> # from above section, how to get node ip to block
+  secret:
+    name: <csi-rbd-provisioner-secret-name>
+    namespace: <ceph-cluster-namespace>
+  parameters:
+    clusterID: <ceph-cluster-namespace>
+```
+
+## Bringing a node back online
+
+Once the node is back online, the admin removes the taint.
+
+1. Remove the taint
+
+```console
+kubectl taint nodes <node-name> node.kubernetes.io/out-of-service=nodeshutdown:NoExecute-
+# or
+kubectl taint nodes <node-name> node.kubernetes.io/out-of-service=nodeshutdown:NoSchedule-
+```
+
+1. Rook will detect the taint is removed from the node, and immediately unfence the node by deleting the corresponding networkFence CR.
+
+## Automation to taint a node
+
+Rook will not automate tainting the node when they go offline. This is a decision the admin needs to make. But Rook will consider creating a sample script to watch for unavailable nodes and automatically taint the node based on how long node is offline. The admin can choose to enable the automated taints by running this example script.
+
+## Open Design Questions
+
+1. How to handle MultiCluster scenarios where two different node on different clusters have the same overlapping IP's.
```

Comment:
The above design is not considering the radosnamespace (required to run rbd status CLI command) and subvolumegroups (for later)

Reply:
I'm missing here, Do you want to add this line or what you are suggesting here?